### PR TITLE
Feature/k8s metadata enhancement

### PR DESCRIPTION
**Pull request by gilltaran**
# PR Submission checklist

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| dell/csm#289|

# Common PR Checklist:

- [x] Have you made sure that the code compiles?
- [x] Have you commented your code, particularly in hard-to-understand areas
- [x] Did you run tests in a real Kubernetes cluster?
- [x] Have you maintained backward compatibility

## Description of your changes:
<added metadata in volume and host>

### CODE REVIEW

**Review by spriya-m — DISMISSED** [review]

File: host_types.go [hunk 1]
```diff
@@ -132,6 +132,8 @@ type HostCreate struct {
 	Name *string `json:"name"`
 	// os type
 	OsType *OSTypeEnum `json:"os_type"`
+
+	Metadata *map[string]string `json:"metadata,omitempty"`
```
Comment (harishp8889) [inline host_types.go:136]: Please add the comment with this format, Otherwise lint checks will fail `// Metadata`

File: volume_types.go [hunk 2]
```diff
@@ -83,6 +83,8 @@ type VolumeCreate struct {
 	// Volume group to add the volume to. If not specified, the volume is not added to a volume group.
 	VolumeGroupID string `json:"volume_group_id,omitempty"`
 
+	Metadata *map[string]string `json:"metadata,omitempty"`
```
Comment (harishp8889) [inline volume_types.go:86]: same here add a comment

**Review by harishp8889 — CHANGES_REQUESTED** [review]

File: host_types.go (hunk 1 above)
Comment (gilltaran) [reply host_types.go:136]: Done

File: volume_types.go (hunk 2 above)
Comment (gilltaran) [reply volume_types.go:86]: done

**Review by spriya-m — DISMISSED** [review]

**Review by spriya-m — APPROVED** [review]

**Review by harshitap26 — APPROVED** [review]

**Review by harishp8889 — APPROVED** [review]

**Pull request merged**